include/range/v3/action/remove.hpp

```diff
@@ -0,0 +1,82 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Andrey Diduh 2019
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+#ifndef RANGES_V3_ACTION_REMOVE_HPP
+#define RANGES_V3_ACTION_REMOVE_HPP
+
+#include <meta/meta.hpp>
+#include <range/v3/core.hpp>
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/action/action.hpp>
+#include <range/v3/action/remove_if.hpp>
+
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-actions
+        /// @{
+        namespace action
+        {
+            struct remove_fn
+            {
+            private:
+                friend action_access;
+                template<typename V, typename P = ident, CONCEPT_REQUIRES_(!Range<V>())>
+                static auto bind(remove_fn remove, const V& value, P proj = P{})
+                RANGES_DECLTYPE_AUTO_RETURN
+                (
+                    std::bind(remove,
+                        std::placeholders::_1,
+                        std::cref(value),
```

Comment:
I'm vaguely uncomfortable with this "capture a reference" formulation, where range-v3 would typically require `value` to be `CopyConstructible` and store it in a `semiregular_t`.

@ericniebler, thoughts? Do you think it's cleaner to enable use of move-only types by capturing references or by relaxing the `Semiregular` requirement for actions?

Reply:
No, you're right. 
I thought about actions as something immobile and applied immediately.
But since     
```cpp
const auto remove_data = action::remove(Data{2});
list |= remove_data;
```
is possible, it is incorrect to stored value as reference.

Reply:
I will replace `std::cref` with `bind_forward`.

But somewhere internally, inside utility/functional.hpp binded function copied (passed by value). I suppose its somewhere in `pipeable::operator|(...)`

At least this prevents move-only types from being used as arguments for actions. _Besides `Semiregular` requirement for actions._